Source/BSGConnectivity.m

```diff
@@ -35,85 +84,51 @@
 static void BSGConnectivityCallback(SCNetworkReachabilityRef target,
                                     SCNetworkReachabilityFlags flags,
                                     void *info) {
-    void (^callbackBlock)(SCNetworkReachabilityFlags) = (__bridge id)(info);
-    callbackBlock(flags);
+    if (bsg_reachability_change_block && BSGConnectivityShouldReportChange(flags)) {
+        BOOL connected = (flags & kSCNetworkReachabilityFlagsReachable);
+        bsg_reachability_change_block(connected, BSGConnectivityFlagRepresentation(flags));
+    }
 }
 
-@interface BSGConnectivity ()
-
-@property(nonatomic, assign) SCNetworkReachabilityRef reachabilityRef;
-@property(nonatomic, strong) dispatch_queue_t serialQueue;
-@property(nonatomic, copy) CallbackBlock callbackBlock;
-
-@end
-
 @implementation BSGConnectivity
 
-- (instancetype)initWithURL:(NSURL *)url
-                changeBlock:(ConnectivityChange)changeBlock {
-    if (self = [super init]) {
-        NSString *hostName = [url host];
-        _reachabilityRef = SCNetworkReachabilityCreateWithName(NULL, [hostName UTF8String]);
-
-        _connectivityChangeBlock = [changeBlock copy];
-        self.serialQueue = dispatch_queue_create("com.bugsnag.cocoa", NULL);
++ (void)monitorURL:(NSURL *)URL usingCallback:(BSGConnectivityChangeBlock)block {
```

Comment:
Would a different name better indicate the purpose of this method (which is not to monitor a URL but to send a breadcrumb *to* a URL).  It's not snappy (but then ObjC seldom is), but maybe something like: `sendConnectivityChangeEventTo:usingCallback:`?